src/shogun/regression/LeastAngleRegression.h

```diff
@@ -218,18 +208,20 @@ class CLeastAngleRegression: public CLinearMachine
 		int32_t &imax, ST& vmax);
 	#endif
 
-private:
-
-	/** Initialize and register parameters */
-	void init();
-
+public:
 	/**
 	* A templated specialization of the train_machine method
 	* @param data training data
 	* @see train_machine
 	*/
 	template <typename ST>
```

Comment:
could you try to make this only available for floating point numbers?
`template <class ST, class = typename std::enable_if<std::is_floating_point<ST>::value>::type>`